backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/network/host/HostSetupNetworksCommand.java

```diff
@@ -340,26 +340,43 @@ protected void executeCommand() {
 
         try (EngineLock monitoringLock = acquireMonitorLock("Host setup networks")) {
             int timeout = getSetupNetworksTimeout();
-            FutureVDSCall<VDSReturnValue> setupNetworksTask = invokeSetupNetworksCommand(timeout);
 
             try {
-                VDSReturnValue retVal = setupNetworksTask.get(timeout, TimeUnit.SECONDS);
-                if (retVal != null) {
-                    if (!retVal.getSucceeded() && retVal.getVdsError() == null && getParameters().rollbackOnFailure()) {
-                        throw new EngineException(EngineError.SETUP_NETWORKS_ROLLBACK, retVal.getExceptionString());
+                boolean networksUpdated = false;
+                boolean succeeded = false;
+
+                if (hasNetworkChanges()) {
+                    //Update networks configuration on the host
+                    FutureVDSCall<VDSReturnValue> setupNetworksTask = invokeSetupNetworksCommand(timeout);
+                    VDSReturnValue retVal = setupNetworksTask.get(timeout, TimeUnit.SECONDS);
+                    if (retVal != null) {
+                        if (!retVal.getSucceeded() && retVal.getVdsError() == null && getParameters().rollbackOnFailure()) {
+                            throw new EngineException(EngineError.SETUP_NETWORKS_ROLLBACK, retVal.getExceptionString());
+                        }
+
+                        VdsHandler.handleVdsResult(retVal);
+
+                        networksUpdated = retVal.getSucceeded();
+                        succeeded = true;
                     }
+                } else {
+                    //We have just changes in labels that should not be reflected on the host itself. So nothing to do
+                    //on the host.
+                    networksUpdated = true;
+                    succeeded = true;
+                }
 
-                    VdsHandler.handleVdsResult(retVal);
-
-                    if (retVal.getSucceeded()) {
-
-                        VDSReturnValue returnValue =
-                                runVdsCommand(VDSCommandType.GetCapabilities,
-                                        new VdsIdAndVdsVDSCommandParametersBase(getVds()));
-                        VDS updatedHost = (VDS) returnValue.getReturnValue();
-                        persistNetworkChanges(updatedHost);
-                    }
+                if (networksUpdated) {
+                    //If host networks were updated successfully then update networks configuration (including changes
+                    //in labels) in the engine DB.
+                    VDSReturnValue returnValue =
+                            runVdsCommand(VDSCommandType.GetCapabilities,
+                                    new VdsIdAndVdsVDSCommandParametersBase(getVds()));
+                    VDS updatedHost = (VDS) returnValue.getReturnValue();
+                    persistNetworkChanges(updatedHost);
+                }
 
+                if (succeeded) {
                     setSucceeded(true);
```

Comment:
`setSucceeded(true)`  can be moved to the `if` above when there is only single boolean.

Reply:
Fixed